sagemaker-python-sdk/tensorflow_script_mode_training_and_serving/tensorflow_script_mode_training_and_serving.ipynb

```diff
@@ -0,0 +1,254 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Set up the environment\n",
+    "\n",
+    "Let's start by setting up the environment:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import sagemaker\n",
+    "from sagemaker import get_execution_role\n",
+    "\n",
+    "sagemaker_session = sagemaker.Session()\n",
+    "eifjccfvcigengfddvtdthlfilvguidttjiugkhdgnvh\n",
+    "\n",
+    "role = get_execution_role()\n",
+    "region = sagemaker_session.boto_session.region_name"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Training Data\n",
+    "\n",
+    "The MNIST dataset has been loaded to public S3 buckets ``sagemaker-sample-data-<REGION>`` under prefix ``tensorflow/mnist``. There are four ``.npy`` file under this prefix:\n",
+    "* ``train_data.npy``\n",
+    "* ``eval_data.npy``\n",
+    "* ``train_labels.npy``\n",
+    "* ``eval_labels.npy``"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "training_data_uri = 's3://sagemaker-sample-data-{}/tensorflow/mnist'.format(region)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Construct a script for distributed training\n",
+    "\n",
+    "This tutorial's training script was adapted from TensorFlow's official [CNN MNIST example](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/tutorials/layers/cnn_mnist.py). We have modified it to handle the ``model_dir`` parameter passed in by SageMaker. This is an S3 path which can be used for data sharing during distributed training and checkpointing and/or model persistence. In addtion we have also added a argument parsing function to handle processing training-related variables.\n",
+    "\n",
+    "At the end of the training job we have added a step to export the trained model to the path stored in the environment variable ``SM_MODEL_DIR`` which always points to ``/opt/ml/model``. This is critical because SageMaker uploads all the model artifacts in this folder to S3 at end of training.\n",
+    "\n",
+    "Here is the entire script:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "!pygmentize 'mnist.py'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Create a training job using the ``sagemaker.tensorflow.TensorFlow`` estimator\n",
```

Comment:
maybe just do `TensorFlow` without the entire path? it's starting to look weird now that the whole thing is there